File: apps/api-gql/internal/delivery/gql/mappers/chat_translation.go
package mappers

import (
	"github.com/twirapp/twir/apps/api-gql/internal/delivery/gql/gqlmodel"
	"github.com/twirapp/twir/apps/api-gql/internal/entity"
)

func ChatTranslationEntityTo(e entity.ChatTranslation) gqlmodel.ChatTranslation {
	return gqlmodel.ChatTranslation{
		ID:                e.ID.String(),
		ChannelID:         e.ChannelID,
		Enabled:           e.Enabled,
		TargetLanguage:    e.TargetLanguage,
		ExcludedLanguages: e.ExcludedLanguages,
		UseItalic:         e.UseItalic,
		ExcludedUsersIDs:  e.ExcludedUsersIDs,
		CreatedAt:         e.CreatedAt,
		UpdatedAt:         e.UpdatedAt,
	}
}

File: web/layers/dashboard/api/chat-translation.ts
import { useQuery } from '@urql/vue'
import { createGlobalState } from '@vueuse/core'

import { graphql } from '@/gql/gql'
import { useMutation } from '~/composables/use-mutation'

const invalidationKey = 'ChatTranslation'

export const useChatTranslationApi = createGlobalState(() => {
	const useQueryChatTranslation = () =>
		useQuery({
			query: graphql(`
      query GetChatTranslation {
        chatTranslation {
          id
          channelID
          enabled
          targetLanguage
          excludedLanguages
          useItalic
          excludedUsersIDs
          createdAt
          updatedAt
        }
      }
    `),
			context: { additionalTypenames: [invalidationKey] },
		})

	const useMutationCreateChatTranslation = () =>
		useMutation(
			graphql(`
    mutation CreateChatTranslation($input: ChatTranslationCreateInput!) {
      chatTranslationCreate(input: $input) {
        id
        channelID
        enabled
        targetLanguage
        excludedLanguages
        useItalic
        excludedUsersIDs
        createdAt
        updatedAt
      }
    }
  `),
			[invalidationKey]
		)

	const useMutationUpdateChatTranslation = () =>
		useMutation(
			graphql(`
    mutation UpdateChatTranslation($id: String!, $input: ChatTranslationUpdateInput!) {
      chatTranslationUpdate(id: $id, input: $input) {
        id
        channelID
        enabled
        targetLanguage
        excludedLanguages
        useItalic
        excludedUsersIDs
        createdAt
        updatedAt
      }
    }
  `),
			[invalidationKey]
		)

	return {
		useQueryChatTranslation,
		useMutationCreateChatTranslation,
		useMutationUpdateChatTranslation,
	}
})
